service/history/historyBuilder.go

```diff
@@ -420,15 +420,24 @@ func (b *historyBuilder) addEventToHistory(event *workflow.HistoryEvent) *workfl
 }
 
 func (b *historyBuilder) newWorkflowExecutionStartedEvent(
-	request *workflow.StartWorkflowExecutionRequest) *workflow.HistoryEvent {
+	startRequest *h.StartWorkflowExecutionRequest, previousRunID *string) *workflow.HistoryEvent {
+	request := startRequest.StartRequest
 	historyEvent := b.msBuilder.createNewHistoryEvent(workflow.EventTypeWorkflowExecutionStarted)
 	attributes := &workflow.WorkflowExecutionStartedEventAttributes{}
 	attributes.WorkflowType = request.WorkflowType
 	attributes.TaskList = request.TaskList
 	attributes.Input = request.Input
 	attributes.ExecutionStartToCloseTimeoutSeconds = common.Int32Ptr(*request.ExecutionStartToCloseTimeoutSeconds)
 	attributes.TaskStartToCloseTimeoutSeconds = common.Int32Ptr(*request.TaskStartToCloseTimeoutSeconds)
+	attributes.ChildPolicy = request.ChildPolicy
+	attributes.ContinuedExecutionRunId = previousRunID
```

Comment:
previous run ID is better than ContinuedExecutionRunId, i assume

Reply:
Same as above.